Remove NooBaa SCCs generation

baum opened the pull request:
Remove NooBaa SCCs generation

Following SCC RBAC PR in Noobaa - https://github.com/noobaa/noobaa-operator/pull/743
- Their operator no longer requires SCC privileges to operate
- All NooBaa SCCs are managed internally by their operator

Signed-off-by: Alexander Indenbaum <aindenba@redhat.com>

baum commented:
/assign @blaineexe

agarwal-mudit commented:
/ok-to-test

baum commented:
/retest

baum commented:
/retest

jarrpa commented:
/override ci/prow/red-hat-storage-ocs-ci-e2e-aws

jarrpa commented:
/test ocs-operator-bundle-e2e-aws

jarrpa commented:
/retest

jarrpa commented:
/retest

jarrpa commented:
/override ci/prow/red-hat-storage-ocs-ci-e2e-aws

jarrpa commented:
/test ocs-operator-bundle-e2e-aws

jarrpa commented:
/test ocs-operator-bundle-e2e-aws

baum commented:
/retest

malayparida2000 commented:
@baum the ocs-operator-ci is a temporary failure, it would pass after several tries.
But the bundle-e2e-aws test failure is mostly a genuine failure.

It's hitting this every time https://github.com/red-hat-storage/ocs-operator/blob/b4d83b21f779f07c4da862e1824eab767440d53d/pkg/deploy-manager/storagecluster.go#L304.

The storagecluster is never reaching the available status, it's getting stuck in progressing stage forever(20 mins is the time of wait here).

This needs further investigation.

malayparida2000 commented:
/retest

baum commented:
Hello @malayparida2000 🖖

I have rebased the PR, and updated `hack/common.sh`

```diff
diff --git a/hack/common.sh b/hack/common.sh
index e8dea978..e2997949 100644
--- a/hack/common.sh
+++ b/hack/common.sh
@@ -44,9 +44,9 @@ NOOBAA_CSV="$OUTDIR_TEMPLATES/noobaa-csv.yaml"
 ROOK_CSV="$OUTDIR_TEMPLATES/rook-csv.yaml.in"
 OCS_CSV="$OUTDIR_TEMPLATES/ocs-operator.csv.yaml.in"

-LATEST_ROOK_IMAGE="rook/ceph:v1.9.9"
-LATEST_NOOBAA_IMAGE="noobaa/noobaa-operator:master-20220527"
-LATEST_NOOBAA_CORE_IMAGE="noobaa/noobaa-core:master-20220527"
+LATEST_ROOK_IMAGE="rook/ceph:v1.9.3"
+LATEST_NOOBAA_IMAGE="noobaa/noobaa-operator:master-20220808"
+LATEST_NOOBAA_CORE_IMAGE="noobaa/noobaa-core:master-20220808"
 LATEST_NOOBAA_DB_IMAGE="centos/postgresql-12-centos7"
 LATEST_CEPH_IMAGE="quay.io/ceph/ceph:v16"
 LATEST_VOLUME_REPLICATION_OPERATOR_IMAGE="csiaddons/volumereplication-operator:v0.1.0"
```

So I have run `make gen-latest-csv`
```
INFO[0001] Bundle metadata generated suceessfully
Manifests sourced into deploy/csv-templates directory
reading in csv at deploy/csv-templates/ocs-operator.csv.yaml.in
reading in csv at deploy/csv-templates/rook-csv.yaml.in
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
main.unmarshalCSV({0x7ffed930021f, 0x25})
	/src/tools/csv-merger/csv-merger.go:358 +0x1109
main.generateUnifiedCSV()
	/src/tools/csv-merger/csv-merger.go:447 +0x5a
main.main()
	/src/tools/csv-merger/csv-merger.go:1074 +0x3ef
make: *** [Makefile:94: gen-latest-csv] Error 2
```

Caused by the following code,  since `len(templateStrategySpec.DeploymentSpecs) == 0`, at this point
```golang

// override the rook env var list.
templateStrategySpec.DeploymentSpecs[0].Spec.Template.Spec.Containers[0].Env = vars

```

Please advice,
~baum

baum commented:
@malayparida2000 hello 🖖

> @baum the ocs-operator-ci is a temporary failure, it would pass after several tries. But the bundle-e2e-aws test failure is mostly a genuine failure.
> 
> It's hitting this every time
> 
> https://github.com/red-hat-storage/ocs-operator/blob/b4d83b21f779f07c4da862e1824eab767440d53d/pkg/deploy-manager/storagecluster.go#L304
> 
> .
> The storagecluster is never reaching the available status, it's getting stuck in progressing stage forever(20 mins is the time of wait here).
> 
> This needs further investigation.

Could you point to the noobaa operator log for the failing test?

malayparida2000 commented:
@iamniting ,The changes are making the storage cluster in the bundle e2e test to be forever stuck in progressing phase, I tried to find the cause but haven't been able to. So can you take a look

iamniting commented:
@baum Did you try to run this locally? If not can you pls try it out and look at the logs where exactly things are going wrong? 

We are not aware of the way to get the logs from the openshift-ci. We tried to get it earlier but there was no success.

baum commented:
> @baum Did you try to run this locally? If not can you pls try it out and look at the logs where exactly things are going wrong?
> 
> We are not aware of the way to get the logs from the openshift-ci. We tried to get it earlier but there was no success.

Local tests pass 😀

To investigate this issue, the NooBaa operator logs during the failing test are required. Is there any way to add those to the test's artifacts?

iamniting commented:
> > @baum Did you try to run this locally? If not can you pls try it out and look at the logs where exactly things are going wrong?
> > We are not aware of the way to get the logs from the openshift-ci. We tried to get it earlier but there was no success.
> 
> Local tests pass grinning
> 
> To investigate this issue, the NooBaa operator logs during the failing test are required. Is there any way to add those to the test's artifacts?

I am not aware of any way around it. you will have to test it with ocs/odf considering you ran the tests with noobaa only.

jarrpa commented:
@iamniting @baum Come on guys, at least try looking a little harder. :stuck_out_tongue_closed_eyes: 

* Pod Logs: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/red-hat-storage_ocs-operator/1699/pull-ci-red-hat-storage-ocs-operator-main-ocs-operator-bundle-e2e-aws/1565348342966259712/artifacts/ocs-operator-bundle-e2e-aws/gather-extra/artifacts/pods/ 
  * noobaa-operator log: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/red-hat-storage_ocs-operator/1699/pull-ci-red-hat-storage-ocs-operator-main-ocs-operator-bundle-e2e-aws/1565348342966259712/artifacts/ocs-operator-bundle-e2e-aws/gather-extra/artifacts/pods/openshift-storage_noobaa-operator-7449987ff4-mwx9d_noobaa-operator.log
  * noobaa-core log: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/red-hat-storage_ocs-operator/1699/pull-ci-red-hat-storage-ocs-operator-main-ocs-operator-bundle-e2e-aws/1565348342966259712/artifacts/ocs-operator-bundle-e2e-aws/gather-extra/artifacts/pods/openshift-storage_noobaa-core-0_core.log

iamniting commented:
Thanks, @jarrpa. Good to know that something like this exists.

malayparida2000 commented:
> @jarrpa thank you for your help 🖖
> 
> Looking at the logs, I can see that despite the update of 'hack/common.sh' see [here](https://github.com/red-hat-storage/ocs-operator/pull/1699/commits/915d089caac6610783961c8a582a90fc346126c3#diff-f6f13ef4e3c83b5e22c436df5b1a4b24fd2c22a80d25855668b41828bcc05e12) still, an old noobaa operator image is used ("noobaa/noobaa-operator:master-20210816" see [here](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/red-hat-storage_ocs-operator/1699/pull-ci-red-hat-storage-ocs-operator-main-ocs-operator-bundle-e2e-aws/1565348342966259712/artifacts/ocs-operator-bundle-e2e-aws/gather-extra/artifacts/pods.json)).
> 
> The older noobaa operator images are expected to fail. Should anything else in addition to hack/common.sh be changed to update the noobaa operator image?

Hi @baum . I found the problem. So in the hack.sh, Along with the 2 places you have made the change for nooba & nooba core images also go a little down to Line 90 and update the nooba-bundle tag also to the latest one which you need, Which for your case should be master-20220808. It should work.

Also remember to remove the merge conflicts & also you changed these tag to master-20220808, But I see we have now master-20220905 available also, So if appropriate update those too.

Thanks

malayparida2000 commented:
> @baum: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> 
> Test name	Commit	Details	Required	Rerun command
> ci/prow/red-hat-storage-ocs-ci-e2e-aws	[631b63e](https://github.com/red-hat-storage/ocs-operator/commit/631b63e1258192eb3966f32e61bff17d7ca7b59a)	[link](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/red-hat-storage_ocs-operator/1699/pull-ci-red-hat-storage-ocs-operator-main-red-hat-storage-ocs-ci-e2e-aws/1564222571602251776)	false	`/test red-hat-storage-ocs-ci-e2e-aws`
> ci/prow/ocs-operator-bundle-e2e-aws	[4b4322f](https://github.com/red-hat-storage/ocs-operator/commit/4b4322f5c76b610a90ef476dd2751ca212176a0a)	[link](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/red-hat-storage_ocs-operator/1699/pull-ci-red-hat-storage-ocs-operator-main-ocs-operator-bundle-e2e-aws/1569255506457923584)	true	`/test ocs-operator-bundle-e2e-aws`
> [Full PR test history](https://prow.ci.openshift.org/pr-history?org=red-hat-storage&repo=ocs-operator&pr=1699). [Your PR dashboard](https://prow.ci.openshift.org/pr?query=is:pr+state:open+author:baum).

This is really weird, I checked out the PR locally & built images & tested. It worked fine. Seems like something is wrong on openshift-ci side on which bundle image they are using. I am digging into it @baum

baum commented:
> This is really weird, I checked out the PR locally & built images & tested. It worked fine. Seems like something is wrong on openshift-ci side on which bundle image they are using. I am digging into it @baum

@malayparida2000, according to the failing test's [pods artifact](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/red-hat-storage_ocs-operator/1699/pull-ci-red-hat-storage-ocs-operator-main-ocs-operator-bundle-e2e-aws/1569255506457923584/artifacts/ocs-operator-bundle-e2e-aws/gather-extra/artifacts/pods.json) "quay.io/**dannyzaken**/noobaa-operator-bundle:**v5.9.0**" bundle image is still used by the CI environment, despite common.sh definition. No reference to that image in the sources. The bundle image could be overridden by the CI environment variable through, see [here](https://github.com/red-hat-storage/ocs-operator/blob/b477367444f56a59f6f68a897d774ddd9c3e3809/hack/common.sh#L92-L95).

Any idea where and how one of the following variables might be defined by the CI environment?
- `NOOBAA_BUNDLE_REGISTRY`
- `NOOBAA_BUNDLE_NAME`
- `NOOBAA_BUNDLE_TAG`

Thank you 🖖!

malayparida2000 commented:
> > This is really weird, I checked out the PR locally & built images & tested. It worked fine. Seems like something is wrong on openshift-ci side on which bundle image they are using. I am digging into it @baum
> 
> @malayparida2000, according to the failing test's [pods artifact](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/red-hat-storage_ocs-operator/1699/pull-ci-red-hat-storage-ocs-operator-main-ocs-operator-bundle-e2e-aws/1569255506457923584/artifacts/ocs-operator-bundle-e2e-aws/gather-extra/artifacts/pods.json) "quay.io/**dannyzaken**/noobaa-operator-bundle:**v5.9.0**" bundle image is still used by the CI environment, despite common.sh definition. No reference to that image in the [ocs-operator sources](https://github.com/red-hat-storage/ocs-operator). The bundle image could be overridden by the CI environment variable through, see [here](https://github.com/red-hat-storage/ocs-operator/blob/b477367444f56a59f6f68a897d774ddd9c3e3809/hack/common.sh#L92-L95).
> 
> Any idea where and how one of the following variables might be defined by the CI environment?
> 
> * `NOOBAA_BUNDLE_REGISTRY`
> * `NOOBAA_BUNDLE_NAME`
> * `NOOBAA_BUNDLE_TAG`
> 
> Thank you 🖖!

I have little prior knowledge of openshift-ci, so I haven't been able to completely understand it. But [here](https://github.com/openshift/release/blob/master/ci-operator/config/red-hat-storage/ocs-operator/red-hat-storage-ocs-operator-main.yaml) is where the tests are defined.

baum commented:
> > > This is really weird, I checked out the PR locally & built images & tested. It worked fine. Seems like something is wrong on openshift-ci side on which bundle image they are using. I am digging into it @baum
> > 
> > 
> > @malayparida2000, according to the failing test's [pods artifact](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/red-hat-storage_ocs-operator/1699/pull-ci-red-hat-storage-ocs-operator-main-ocs-operator-bundle-e2e-aws/1569255506457923584/artifacts/ocs-operator-bundle-e2e-aws/gather-extra/artifacts/pods.json) "quay.io/**dannyzaken**/noobaa-operator-bundle:**v5.9.0**" bundle image is still used by the CI environment, despite common.sh definition. No reference to that image in the [ocs-operator sources](https://github.com/red-hat-storage/ocs-operator). The bundle image could be overridden by the CI environment variable through, see [here](https://github.com/red-hat-storage/ocs-operator/blob/b477367444f56a59f6f68a897d774ddd9c3e3809/hack/common.sh#L92-L95).
> > Any idea where and how one of the following variables might be defined by the CI environment?
> > 
> > * `NOOBAA_BUNDLE_REGISTRY`
> > * `NOOBAA_BUNDLE_NAME`
> > * `NOOBAA_BUNDLE_TAG`
> > 
> > Thank you 🖖!
> 
> I have little prior knowledge of openshift-ci, so I haven't been able to completely understand it. But [here](https://github.com/openshift/release/blob/master/ci-operator/config/red-hat-storage/ocs-operator/red-hat-storage-ocs-operator-main.yaml) is where the tests are defined. You can also take a look.

@malayparida2000 thanks for the release repo link. Does not look that "quay.io/**dannyzaken**/noobaa-operator-bundle:**v5.9.0**"  bundle image name is define there.

@dannyzaken @jarrpa any idea where "quay.io/**dannyzaken**/noobaa-operator-bundle:**v5.9.0**"  definition comes from?

BlaineEXE commented:
I don't see where the `quay.io/dannyzaken/noobaa-operator-bundle:v5.9.0` is coming from in this repo after searching for `dannyzaken`. Perhaps it's defined in another project's bundle? 

I also didn't see it searching for `dannyzaken` in the noobaa repo.

I'm also not seeing it in github.com/openshift/release, and there I used my better local search tools.

I'm not sure where else to look.

malayparida2000 commented:
Hello @baum , Hopefully i have this figured finally. Just run
    ./openshift-ci/build-operator-index-ci.sh
You will see a index.db file changed. Just add that to your PR.
this can finally fix this I hope.

Basicaly openshift-ci fetches the nooba bundle to use from this index.db file.
That's where the old image was defined, And as it's inside the db file we don't get the image when we try to find it by name.
@jarrpa Thanks for the help in understanding this.

baum commented:
Hello @malayparida2000  @jarrpa 🖖
Thank you for your help!

baum commented:
Added [Verify the CI index database is up to date](https://github.com/red-hat-storage/ocs-operator/pull/1820) PR, to be added to [CI operator main workflow](https://github.com/openshift/release/blob/master/ci-operator/config/red-hat-storage/ocs-operator/red-hat-storage-ocs-operator-main.yaml), similar to verify-latest-csv

iamniting commented:
Hello @baum your changes look good to me, Can you pls resolve conflict?

iamniting reviewed (approved):
/lgtm

malayparida2000 commented on controllers/ocsinitialization/sccs.go:
```diff
@@ -46,8 +45,6 @@ func getAllSCCs(namespace string) []*secv1.SecurityContextConstraints {
-		newNooBaaEndpointSCC(namespace),
 	}
 }
 
```
@baum The blankSCC function also needs to

malayparida2000 commented:
/lgtm
